packages/cli/src/agentInstaller/pythonAgentInstaller.js

```diff
@@ -0,0 +1,165 @@
+/* eslint-disable class-methods-use-this */
+/* eslint-disable max-classes-per-file */
+// @ts-check
+
+const { promises: fsp } = require('fs');
+const { spawn } = require('child_process').spawn;
+const BuildToolInstaller = require('./buildToolInstallerBase');
+const CommandStruct = require('./commandStruct');
+const InstallAgentStep = require('./installAgentStep');
+const ValidationError = require('../errors/validationError');
+
+/**
+ * @typedef {import('./types').AgentInstaller} AgentInstaller
+ * @typedef {import('./types').Command} Command
+ * @typedef {import('./types').InstallResult} InstallResult
+ * @typedef {import('./types').InstallStep} InstallStep
+ */
+
+const POETRY_LOCK_FILE = 'poetry.lock';
+const REQUIREMENTS_FILE = 'requirements.txt';
+
+const REGEX_PKG_DEPENDENCY = /^appmap.*?=.*$/m;
+// include .dev0 so pip will allow prereleases
+const PKG_DEPENDENCY = 'appmap>=1.1.0.dev0';
+
+class PoetryInstaller extends BuildToolInstaller {
+  /**
+   * @param {string} path
+   */
+  constructor(path) {
+    super(POETRY_LOCK_FILE, path);
+  }
+
+  /**
+   * @returns {string}
+   */
+  get assumptions() {
+    return `Your project contains a ${POETRY_LOCK_FILE}. Therefore, it looks like a poetry project,
+so we will install the AppMap Python project "appmap".`;
```

Comment:
Do you mean "package" here instead of "project"?

Reply:
Yes. Fixed.